reproducible/README.md

```diff
@@ -0,0 +1,343 @@
+   user and group information.  We default to "azureuser" "1000" "/home/azureuser"
+
+The script "build-oe-nix-image.sh" is a thin wrapper. It takes no arguments, and just passes the needed arguments
+to docker build.  If it needs to be modified, a new script should be created.
+
+Dockerfile.nix prepares the build image by:
+
+- adding via apt minimum packages for the outer image. This includes a text editor (vim) and ELF patch utility (patchelf).
+
+- sets up nit configuration.
+
+- diables apt.
+
+- creates the build user and directories, groups etc.
+
+- creates a nix expression for building openenclave via (shell.nix).
+
+- installs nix as the build user
+
+- calls the prep-nix-build.sh script which sets up the git clone of nix pkgs and preinstalls the package
+  derivations needed for building openenclave into the nix store. The search process for dependencies in
+  nix is to refer to the nix store first, before looking at nixpkgs or contacting the cache if enabled. While
+  we cannot select package versions from the expression, but must depend on the nixpkgs repo to provide the
+  correct versions, we can prime the nix store to provide the exact derivation we wish before referring to the repo.
+
+  These derivations have been manually chosen.  This also allows us to set package priorities which is required
+  between packages that provide overlapping contents, for example clang_7 and llvm_7.
+
+- adds a build.env file which contains environment values to control the build.  This is not part of the 
+  - REV is the git revision (commit or tag) to pull for the build.
+  - SHA is the sha256 hash of the tar file downloaded from git.
+  - DO_CHECK enables ctests as part of the build.
+  - DEB_SHA specifies the expected result sha25sum for the .deb file produced. An error will be thrown if the 
+    sha is incorrect. If not set, the sha is not checked.
+
+- adds needed files for ctest to the .nix_libs directory, which will be referred to by ctest. Nix expressions
+  cannot refer to libraries outside the minimum environment and explicit inputs, so these libraries are needed
+  to use sgx (and eventually TrustZone) facilties.
+
+- sets the build user as the user. Among other things, this means the build image when run cannot be 
+  modified by the addition of apt packages.
+
+The resulting image is named openenclave-build. It can (and should) be generated once and pushed into a common 
+repo.
+
+### build_openenclave.sh
+
+Runs the image produced by docker build, with arguments resolved.
+The output of the process is a .deb and a .nar file containing an exported nix store for use in developing 
+openenclave sdk apps using nix.
+
+
+### workflow
+
+```
+   ./build-oe-nix-image.sh
+```
+will build the nix docker image. This can be pushed:
+```
+   docker push openenclave-build
+```
+and later pulled, or just used locally.
+
+To build the openenclave sdk 
+```
+   ./build_openenclave.sh -o ./result
```

Comment:
```suggestion
   ./build_openenclave.sh -o result
```
As the output directory is being used as a docker volume mount, the leading `.` would cause docker to freak out.

Seeing error similar to -
```
$ docker run -it -v ./result:/tmp/ hello-world
docker: Error response from daemon: create ./result: "./result" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
```

Reply:
Confirmed. Fix in a minute. I've been using ~/result without trouble, but that appears to be an absolute path.

Reply:
fixed. Thank you.